### PR TITLE
Fix ordering of spack compiler find, and move installs to separate layer

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: ubuntu/clang/Dockerfile [hunk 1]
```diff
@@ -5,6 +5,6 @@ FROM ghcr.io/rse-radiuss/ubuntu:$ubuntu_version
 ARG llvm_version
 ENV llvm_version=$llvm_version
 
-RUN spack install llvm@${llvm_version} && \
-  spack compiler find && \
-  spack view --dependencies no symlink --ignore-conflicts /opt/view llvm@${llvm_version}
+RUN spack install llvm@${llvm_version}
```
Comment (vsoch) [inline ubuntu/clang/Dockerfile:8]: Let's do without debug so it's smaller.
```suggestion
RUN spack install llvm@${llvm_version} build_type=Release
```

**Comment by vsoch** [comment]
Do you want to add the caching to this PR (or are you doing it separately?)

File: ubuntu/clang/Dockerfile (hunk 1 above)
Comment (davidbeckingsale) [reply ubuntu/clang/Dockerfile:8]: That's the default: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/llvm/package.py#L135

Comment (vsoch) [reply ubuntu/clang/Dockerfile:8]: oh that's strange, I guess maybe the default used to be debug and now it's Release then! This comment must be old https://github.com/spack/spack/blob/2ea1bf84ece3175c78a0474a3d53b6814ba4850c/var/spack/repos/builtin/packages/llvm/package.py#L62

**Pull request merged**